ext/js/application.js

```diff
@@ -193,6 +167,45 @@ export class Application extends EventDispatcher {
         this.trigger('closePopups', {});
     }
 
+    /**
+     * @param {(application: Application) => (Promise<void>)} mainFunction
+     */
+    static async main(mainFunction) {
```

Comment:
Hmm, I've never really seen this pattern before. I'm more used to have a `create` or `initialize` static function which does the more complex initialization for a class that's not appropriate in its constructor, and then actually return that class. 
Obviously, your way saves a few extra lines compared to that, since we can also deduplicate the try/catch/finally. Did you make that tradeoff and decide to go with maximal deduplication? (And/or is this a fairly common pattern which I'm just unaware of?)

Reply:
It's not really a common pattern, you are correct. The intent is to reduce code duplication for each main function, otherwise each main basically needs to do this:

```js
        const webExtension = new WebExtension();
        const api = new API(webExtension);
        await this.waitForBackendReady(webExtension);
        const {tabId = null, frameId = null} = await api.frameInformationGet();
        const crossFrameApi = new CrossFrameAPI(api, tabId, frameId);
        crossFrameApi.prepare();
        const application = new Application(api, crossFrameApi);
        application.prepare();
        try {
            // rest of code
        } catch (error) {
            log.error(error);
        } finally {
            application.ready();
        }
```

Some of the previous main functions didn't have all of this code, such as the error catching, probably just because I was lazy when writing those. Same for the missing application.ready() in some places.

This function could technically reside outside of the `Application` class, and I could do that if you think that would make things more clean.

My rationale behind this is that the improved consistency and duplicate code reduction across all files is more desirable.

The rationale behind moving `frameInformationGet` out is that there are more places that would benefit from having this globally known at the time of application init. `ext/js/pages/settings/popup-preview-frame-main.js` for example will redundantly re-request these values. (I did not address this in this PR to keep the scope limited, but I would like to address this further afterwards.)

The rationale for moving api out is so that frameInformationGet can be performed prior to the application constructor.

For both api and crossFrameApi, the additional rationale was to allow these values to be fully-prepared non-nullable fields inside of the Application class.